exporter/kafkaexporter/pdata_marshaler.go

```diff
@@ -14,30 +14,68 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil"
 )
 
+// KeyableLogsMarshaler is an extension of the LgosMarshaler interface intended to provide partition key capabilities
+// for log messages
+type KeyableLogsMarshaler interface {
+	LogsMarshaler
+	Key()
+}
+
 type pdataLogsMarshaler struct {
```

Comment:
If pdata is the only one supporting this, can we make sure to mention this in the README?

Reply:
Updated the readme in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33633/commits/70ee65384fc17053869e0e9cbbed143b49ed3035